### ISSUE TITLE
Tram interferes with normal operation of intersection

### ISSUE THREAD

**Issue by HarryBotter2333** [slyh/Cities2-TrafficLightsEnhancement#82]
![屏幕截图 2024-02-09 082051](https://github.com/slyh/Cities2-TrafficLightsEnhancement/assets/43106121/66c34642-fa10-4b93-b1a8-7776425ce263)

The tram interferes with the normal operation of the intersection. It can be seen that the tram has the highest priority. Once the tram approaches the intersection, it will let the tram go first. However, the order of the intersection is destroyed and the traffic rate of the intersection is affected.

**Comment by E2clipse** [comment]
A tram always gets priority.

**Comment by Bendak** [comment]
Indeed. This is by design. And if you look in YouTube, there is the tram system where it gets the tram design and show that tram remotely changes the lights so it can pass.

The problem however is that sometimes, trams disappear because there was a pedestrian crossing in front of it, popping over 200 people in the middle of the tram rails, and sometimes it makes like 2, 5, 10 in a row to disappear. This is an issue with many transport systems. So, any crossing, sometime start to get heavy jams if you have a line of trans waiting to go by unless you have vanilla mode or advanced split phasing where it reduces the amount of time for the full rotation of ligths and don't use the pedestrian phase, it will reduce the issues caused by trams.

For this mod specifically, if there is a way to detect tram, give it priority and after it passes, just revert to the last phase it was it, it would be great as the issue is that the phase rotation resets to the phase of the tram line. And many times, we cannot choose advanced split phasing, only vanilla/split phasing is available when you lay down tram tracks in a crossing.
